### DISCUSSION TITLE
Processing Files After Upload.|| use FFMPEG on server side

### DISCUSSION THREAD

**Discussion by mukhtharcm** [pocketbase/pocketbase#4948] [Q&A]
Hi,
I'm creating an application which involves recording and uploading audio. I can only record audio in webm format and want to convert to ogg format.
Can I convert webm to ogg upon uploading to pocketbase?

TLDR: I want to use ffmpeg on the server side upon uploading a File

**Comment by ganigeorgiev** [comment]
There is no easy solution for intercepting uploaded files at the moment. Record file operations will be simplified with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) but for now you could check the code sample from https://github.com/pocketbase/pocketbase/discussions/3315.

**Comment by thangngoc89** [comment]
Since these kind of processing sounds like resources intensive. You should push the jobs to a queue and have workers listen to and interact with pocketbase using API.

**Comment by lukepighetti** [comment]
For the time being I would set up a separate process to check for new uploads and perform any converting. This will allow you to move this process to a dedicated server as needed and scale independently.

**Reply by mukhtharcm** [reply]
like a cron job that fetches all the new data every few minutes?

**Reply by thangngoc89** [reply]
> like a cron job that fetches all the new data every few minutes?

You should look into queues. 
asynq would be a good start for golang 

https://github.com/hibiken/asynq

**Comment by brvier** [comment]
I do similar job, which identify songs on a record, then divide an audio file in multiple part. I just added a processed boolean to my files table default to false.

Then added a scheduler which do the job in backgroud
```
	app.OnAfterBootstrap().Add(func(e *core.BootstrapEvent) error {
		scheduler := cron.New()
	
		scheduler.MustAdd("identifyRecords", "*/5 * * * *", func() {
			identifyRecords(app)
		})
```

```	
func identifyRecords(app core.App) {

records, err := app.Dao().FindRecordsByFilter(
		"records",         // collection
		"processed = 0", // filter
		"timestamp",       // sort
		10,                // limit
		0,                 // offset
		dbx.Params{},
	)
	if err != nil {
		return
	}

	for _, record := range records {
               
		file, err := os.Open(filepath.Join("pb_data", "storage", record.BaseFilesPath(), record.GetString("audio")))
		if err != nil {
			app.Logger().Error("Opening audio file failed",
				"records.id", record.Id,
				"err", err)
			continue
		}

               [...]
               
               createExtract(app, record, time.Duration(12*time.Second), time.Duration(16*time.Second))
               
func createExtract(app core.App, record *models.Record, start time.Duration, end time.Duration) (filePath string, err error) {

	inPath := filepath.Join("pb_data", "storage", record.BaseFilesPath(), record.GetString("audio"))
	u, _ := uuid.NewRandom()
	outPath := filepath.Join("/tmp", fmt.Sprintf("%s.aac", u))
	// Build the ffmpeg command
	cmd := exec.Command(
		"ffmpeg",
		"-i", inPath,
		"-ss", fmt.Sprint(int(start.Seconds())),
		"-to", fmt.Sprint(int(end.Seconds())),
		"-vn",             // Disable video stream
		"-acodec", "copy", // Copy audio stream without re-encoding
		"-y",    // Overwrite if output file exists
		outPath, // Output filename
	)
	app.Logger().Debug("createExtract", "start", start.Seconds(), "end", end.Seconds())

	// Execute the command
	output, err := cmd.CombinedOutput()
	if err != nil {
		return "", fmt.Errorf("failed to extract audio: %w\n%s", err, string(output))
	}

	// Return the temporary audio file path
	return outPath, nil

}


```
